### PR TITLE
Travis minikube

### PR DESCRIPTION
**Pull request by leafty**
Duplicate of #331, but from here to enable secrets in travis build

Merge after #353

### CODE REVIEW

**Comment by leafty** [comment]
We're getting close:
```
$ kubectl -n renku logs -f renku-test
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-3.6.3, py-1.5.4, pluggy-0.6.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /tests, inifile:
plugins: splinter-1.9.1
collecting ... collected 3 items

test_apispec.py::test_swagger_paths SKIPPED                              [ 33%]
test_renku.py::test_renku_login PASSED                                   [ 66%]
test_renku.py::test_notebook_launch FAILED                               [100%]

=================================== FAILURES ===================================
_____________________________ test_notebook_launch _____________________________

browser = <splinter.driver.webdriver.firefox.WebDriver object at 0x7f3824efe438>

    def test_notebook_launch(browser):
        """Test launching a notebook from UI."""
        url = urljoin(os.getenv('RENKU_ENDPOINT', 'http://localhost'), '/login')
        browser.visit(url)
    
        browser.fill('username', 'cramakri')
        browser.fill('password', 'cramakri')
        browser.find_by_id('kc-login').click()
        assert 'Renku' in browser.title
    
        # go to the project page
>       assert browser.is_element_present_by_text(
            'cramakri/weather-zh', wait_time=10
        )
E       AssertionError: assert False
E        +  where False = <bound method BaseWebDriver.is_element_present_by_text of <splinter.driver.webdriver.firefox.WebDriver object at 0x7f3824efe438>>('cramakri/weather-zh', wait_time=10)
E        +    where <bound method BaseWebDriver.is_element_present_by_text of <splinter.driver.webdriver.firefox.WebDriver object at 0x7f3824efe438>> = <splinter.driver.webdriver.firefox.WebDriver object at 0x7f3824efe438>.is_element_present_by_text

test_renku.py:50: AssertionError
================ 1 failed, 1 passed, 1 skipped in 25.03 seconds ================
```

**Comment by rokroskar** [comment]
wow the tests actually take much less time to run -- did you eliminate something?

**Comment by leafty** [comment]
No, I did not touch it

**Comment by leafty** [comment]
@rokroskar did the cleanup

File: .travis.yml [hunk 1]
```diff
@@ -26,8 +27,9 @@ services:
 
 env:
   global:
-    - HELM_URL=https://storage.googleapis.com/kubernetes-helm
-    - HELM_TGZ=helm-v2.9.1-linux-amd64.tar.gz
+    - DOCKER_HOST="tcp://86.119.32.161:443"
```
Comment (rokroskar) [inline .travis.yml]: Do we want to encrypt this?

Comment (leafty) [reply .travis.yml]: Open ports are secured:
```
ALLOW IPv4 443/tcp from 0.0.0.0/0
ALLOW IPv4 8443/tcp from 0.0.0.0/0
ALLOW IPv4 22/tcp from 0.0.0.0/0
```

- 443: Docker port, secured with TLS auth
- 8443: Kubernetes API port, secured with TLS auth
- 22: ssh, ssh-key only

File: scripts/run-tests.sh [hunk 2]
```diff
@@ -23,9 +23,23 @@ set -o errexit
 # quit on unbound symbols:
 set -o nounset
 
-docker images --digests
+set -ex
 
-pytest -v
+#docker images --digests
```
Comment (rokroskar) [inline scripts/run-tests.sh]: remove the comment

File: scripts/run-tests.sh [hunk 3]
```diff
@@ -23,9 +23,23 @@ set -o errexit
...
+sleep 5 # can help
+kubectl -n $RENKU_DEPLOY run renku-demo -it \
+--env="GITLAB_URL=http://$MAXIKUBE_HOST:32080/gitlab" \
+--env="KEYCLOAK_URL=http://$MAXIKUBE_HOST:32080" \
+--image=renku/renku-demo:latest \
+--restart=Never
+
+helm test $RENKU_DEPLOY
+
+# pytest -v
```
Comment (rokroskar) [inline scripts/run-tests.sh]: remove

File: scripts/travis-before-script.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    helm delete --purge $release
+done
+
+helm init --wait
+helm upgrade --install nginx-ingress --namespace kube-system \
+    --set controller.service.type=NodePort \
+    --set controller.service.nodePorts.http=32080 \
+    --set controller.service.nodePorts.https=32443 \
+    stable/nginx-ingress
+# # This makes sure the ssh nodePort is set to 32022
```
Comment (rokroskar) [inline scripts/travis-before-script.sh]: remove?

File: scripts/travis-install.sh [hunk 5]
```diff
@@ -28,8 +28,26 @@ set -ex
 # install graphviz
 sudo apt-get update && sudo apt-get install -y graphviz
 
-# install geckodriver
-wget https://github.com/mozilla/geckodriver/releases/download/v0.19.1/geckodriver-v0.19.1-linux64.tar.gz
-tar -xvzf geckodriver*
-chmod +x geckodriver
-sudo mv geckodriver /usr/local/bin
+# # install geckodriver
```
Comment (rokroskar) [inline scripts/travis-install.sh]: remove commented section

File: .travis.yml [hunk 6]
```diff
@@ -57,31 +54,28 @@ python:
...
+  - ./scripts/travis-build.sh
+  - ./scripts/run-tests.sh
 
 deploy:
   - provider: script
     skip_cleanup: true
     script: bash scripts/travis-deploy.sh
     on:
       branch: master
-      condition: $TRAVIS_EVENT_TYPE != cron
```
Comment (rokroskar) [inline .travis.yml]: is there a reason why this was removed here?

Comment (leafty) [reply .travis.yml]: But it works now! :rofl: 
I guess the PR was based before the line was introduced

Comment (rokroskar) [reply .travis.yml]: I just noticed lots of pushes to the helm charts at a 24-hour cadence...

**Pull request merged**